application/src/main/java/org/opentripplanner/routing/algorithm/mapping/ViaLocationTypeMapper.java

```diff
@@ -0,0 +1,53 @@
+package org.opentripplanner.routing.algorithm.mapping;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.opentripplanner.model.plan.leg.ViaLocationType;
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.street.model.vertex.TemporaryStreetLocation;
+import org.opentripplanner.transit.model.site.StopLocation;
+
+/**
+ * Maps locations to {@link ViaLocationType} based on their existence in the route request.
+ */
+public class ViaLocationTypeMapper {
+
+  @Nullable
+  public static ViaLocationType map(RouteRequest request, StopLocation stop) {
```

Comment:
You're passing in a very large object (`RouteRequest`) but you only need a very small part of it. Is this a good idea?